store/sqlstore/bot_store.go

```diff
@@ -0,0 +1,246 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+package sqlstore
+
+import (
+	"database/sql"
+	"net/http"
+	"strings"
+
+	"github.com/mattermost/mattermost-server/einterfaces"
+	"github.com/mattermost/mattermost-server/model"
+	"github.com/mattermost/mattermost-server/store"
+)
+
+// bot is a subset of the model.Bot type, omitting the model.User fields.
+type bot struct {
+	UserId      string `json:"user_id"`
+	Description string `json:"description"`
+	CreatorId   string `json:"creator_id"`
+	CreateAt    int64  `json:"create_at"`
+	UpdateAt    int64  `json:"update_at"`
+	DeleteAt    int64  `json:"delete_at"`
+}
+
+func botFromModel(b *model.Bot) *bot {
+	return &bot{
+		UserId:      b.UserId,
+		Description: b.Description,
+		CreatorId:   b.CreatorId,
+		CreateAt:    b.CreateAt,
+		UpdateAt:    b.UpdateAt,
+		DeleteAt:    b.DeleteAt,
+	}
+}
+
+// SqlBotStore is a store for managing bots in the database.
+// Bots are otherwise normal users with extra metadata record in the Bots table. The primary key
+// for a bot matches the primary key value for corresponding User record.
+type SqlBotStore struct {
```

Comment:
I notice you are using the "old style" store to implement this. I was under the impression that new stores were meant to use the "new style" `supplier` interfaces, or has that effort been abandoned?

Reply:
I think I missed the "new style" `supplier` memo, as everything I've touched to date seems to have been in the "old style". Can you point me in the right direction for the latest on this topic? I'm happy to effect the required changes, though less happy if that requires substantial unit test rewrites :|.

Reply:
Not 100% sure about this, but looping in @crspeller as he introduced the "supplier" based store architecture.